examples/extension/build.gradle

```diff
@@ -92,17 +92,43 @@ dependencies {
   otel("io.opentelemetry.javaagent:opentelemetry-javaagent:${versions.opentelemetryJavaagent}:all")
 }
 
+//Extracts manifest from OpenTelemetry Java agent to reuse it later
+task agentManifest(type: Copy) {
+  from zipTree(configurations.otel.singleFile).matching {
+    include 'META-INF/MANIFEST.MF'
+  }
+  into buildDir
+}
+
+//Produces a copy of upstream javaagent with this extension jar included inside it
+//The location of extension jar inside agent jar is hard-coded in the agent source code
+task extendedAgent(type: Jar) {
+  dependsOn agentManifest
+  archiveFileName = "opentelemetry-javaagent-all.jar"
+  from zipTree(configurations.otel.singleFile)
+  from tasks.shadowJar.archiveFile
+  manifest.from "$buildDir/META-INF/MANIFEST.MF"
+  rename { String f ->
+    if (f == tasks.shadowJar.archiveFileName.get()) {
+      return "otel-extensions.jar"
+    } else {
+      return f
+    }
+  }
+}
+
 tasks {
   test {
     useJUnitPlatform()
 
-    def extensionJar = tasks.shadowJar
+    def extensionJar = tasks.extendedAgent
```

Comment:
Should we have two test cases/tasks here? One for extension placed inside the javaagent jar, one for external one

Reply:
Yeah... Wanted to get some feedback before I start messing with gradle test tasks

Reply:
Still one test task and test class, but now we actually test both with extended agent and the original one.